hera_cal/tests/test_io.py

```diff
@@ -0,0 +1,283 @@
+            uvd2.unphase_to_drift()
+        uvd = uvd1 + uvd2
+        d, f = io.load_vis([uvd1,uvd2],nested_dict=True)
+        for i, j in d:
+            for pol in d[i, j]:
+                uvpol = list(uvd1.polarization_array).index(str2pol[pol])
+                uvmask = np.all(
+                    np.array(zip(uvd.ant_1_array, uvd.ant_2_array)) == [i, j], axis=1)
+                np.testing.assert_equal(d[i, j][pol], np.resize(
+                    uvd.data_array[uvmask][:, 0, :, uvpol], d[i, j][pol].shape))
+                np.testing.assert_equal(f[i, j][pol], np.resize(
+                    uvd.flag_array[uvmask][:, 0, :, uvpol], f[i, j][pol].shape))
+
+        d, f = io.load_vis([filename1, filename2],nested_dict=True)
+        for i, j in d:
+            for pol in d[i, j]:
+                uvpol = list(uvd.polarization_array).index(str2pol[pol])
+                uvmask = np.all(
+                    np.array(zip(uvd.ant_1_array, uvd.ant_2_array)) == [i, j], axis=1)
+                np.testing.assert_equal(d[i, j][pol], np.resize(
+                    uvd.data_array[uvmask][:, 0, :, uvpol], d[i, j][pol].shape))
+                np.testing.assert_equal(f[i, j][pol], np.resize(
+                    uvd.flag_array[uvmask][:, 0, :, uvpol], f[i, j][pol].shape))
+
+    
+        uvd = UVData()
+        uvd.read_miriad(filename1)
+        self.assertEqual(len(io.load_vis([uvd],nested_dict=True)[0]), uvd.Nbls)
+        # reorder baseline array
+        uvd.baseline_array = uvd.baseline_array[np.argsort(uvd.baseline_array)]
+        self.assertEqual(len(io.load_vis(filename1,nested_dict=True)[0]), uvd.Nbls)
+
+    #TODO: implement this test
+    def test_write_vis(self):
+        with self.assertRaises(NotImplementedError):
+            io.write_vis(None, None, None)
+
+    
+    def test_update_vis(self):
+        # load in cal
+        fname = os.path.join(DATA_PATH, "zen.2458043.12552.xx.HH.uvORA")
+        outname = os.path.join(DATA_PATH, "test_output/zen.2458043.12552.xx.HH.modified.uvORA")
+        uvd = UVData()
+        uvd.read_miriad(fname)
+        data, flags, antpos, ants, freqs, times, lsts, pols = io.load_vis(fname, return_meta=True)
+
+        #make some modifications
+        new_data = {key: (2.+1.j)*val for key,val in data.items()}
+        new_flags = {key: np.logical_not(val) for key,val in flags.items()}
+        io.update_vis(fname, outname, data=new_data, flags=new_flags,
+                      add_to_history='hello world', clobber=True, telescope_name='PAPER')
+        
+        #test modifications
+        data, flags, antpos, ants, freqs, times, lsts, pols = io.load_vis(outname, return_meta=True)
+        for k in data.keys():
+            self.assertTrue(np.all(new_data[k] == data[k]))
+            self.assertTrue(np.all(new_flags[k] == flags[k]))
+        uvd2 = UVData()
+        uvd2.read_miriad(outname)
+        self.assertEqual(uvd2.history.replace('\n',''), ('hello world' + uvd.history).replace('\n',''))
```

Comment:
FYI, pyuvdata has a function `utils.check_histories` which does these replacements (and one for spaces which can also be a problem).

Reply:
Good call. Nice to see we arrived at the same solution independently.